learn/components/styling.md

```diff
@@ -3,4 +3,191 @@ title: Styling
+    or styled since it’s featureless.
+- A **shadow host** is not a part of a **shadow tree**. It exists
+  _outside_ of a shadow tree and therefore cannot be selected directly
+  by styles declared within its shadow tree (i.e. referencing the tag
+  name with a type selector will not select the host — only nested
+  elements of that type). We can select it with the `:host`
+  pseudo-class and its functional counterpart `:host()` — more on that
+  later.
+  - `:host` when combined with relative selectors will be based on the
+    shadow root within the shadow tree and not in the light tree.
+- Being a part of the DOM outside of a shadow tree, the host element
+  can be subject to styles declared in that context.
+
+### CSS Inheritance
+
+- **A shadow host’s styles become the point of inheritance for all
+  top-level elements within a shadow tree.**
+  - For example, setting `font-family: sans-serif` for the `html`
+    element in a document-level stylesheet can affect our Web
+    Component since `font-family` is generally an inherited property
+    for most elements, including an autonomous custom element by default.
+  - Inheritance is one way that styles from an oustide DOM can enter
+    into a shadow tree.
+
+#### Custom Properties
+
+- CSS Custom Properties inherit by default.
+  - _Could be a subheading_
+  - There is an upcoming feature to CSS which allows CSS authors to
+    configure whether a custom property will inherit by default (see
+    `@property`).
+
+## The Order of Precedence
+
+### Kinda like user-agent stylesheets...
+
+  - Styles set inside of a shadow tree work similar to a user-agent
+    stylesheet.
+    - For example, inside a shadow tree you can set your custom
+      element’s default styles using the `:host` pseudo-class. If there
+      are styles declared for the element outside of the shadow tree,
+      those will win. It might seem counter intuitive at first,
+      especially when you compare the strength of the specificity.
+  - **Specificity does not matter between DOM trees.**
+    - `:host` technically has a specificity value of 0,1,0.
+    - `fancy-button` has a specificity score of 0,0,1,
+    - The latter wins even if you wrap it with `:where()` which
+      decreases it to 0,0,0.
+
+### Using `!important` to prevent outside styling
+
+The only way to completely prevent shadow tree styles from being
+overriden is to use `!important`.
+
+```html
+<style>
+  fancy-button {
+    font-size: 2rem; /* Wins */
+    color: deeppink; /* Loses */
+    font-family: serif !important; /* Loses */
+  }
+</style>
+<fancy-button>
+  <template shadowroot="open">
+    <style>
+      :host {
+        font-size: 1rem; /* Loses */
+        color: dodgerblue !important; /* Wins */
+        font-family: sans-serif !important; /* Wins */
+      }
+    </style>
+    <slot></slot>
+    Shadow text
+  </template>
+  Slotted Text
+</fancy-button>
+```
+
+## Slots
```

Comment:
Slots may deserve their entire own section, which  CSS sections can then refer to